controllers/sriovnetwork_controller.go

```diff
@@ -141,6 +146,14 @@ func (r *SriovNetworkReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	err = r.Get(ctx, types.NamespacedName{Name: netAttDef.Name, Namespace: netAttDef.Namespace}, found)
 	if err != nil {
 		if errors.IsNotFound(err) {
+			targetNamespace := &corev1.Namespace{}
+			err = r.Get(ctx, types.NamespacedName{Name: netAttDef.Namespace}, targetNamespace)
+			if errors.IsNotFound(err) {
+				// Target namespace doesn't exist, avoid retrying with exponential backoff
+				reqLogger.Info("Target namespace doesn't exist, cannot create NetworkAttachmentDefinition", "Namespace", netAttDef.Namespace, "Name", netAttDef.Name)
+				return reconcile.Result{}, nil
```

Comment:
a less elegant solution is to just reconcile after X seconds using Result..RequeueAfter

this solution is much better !